packages/ui/src/components/ManageChannels/ManageChannelsScreenView.tsx

```diff
@@ -330,9 +330,40 @@ export function ManageChannelsScreenView({
         type: c.type,
       })),
     }));
-
-    // Only update if the total number of sections have changed
+    const currentTotalChannels = sections.reduce(
+      (acc, section) => acc + section.channels.length,
+      0
+    );
+    const newTotalChannels = newNavSections.reduce(
+      (acc, section) => acc + section.channels.length,
+      0
+    );
+
+    // Only update local state if the total number of sections have changed
+    // OR if a newly created channel has been added to the default section
+    // OR if a channel has been deleted
     if (newNavSections.length === sections.length) {
+      if (newTotalChannels !== currentTotalChannels) {
+        // Check if a new channel has been added to the default section
+        if (
+          newNavSections.some(
+            (s) =>
+              s.id === 'default' &&
```

Comment:
Why do we only check if new channels were added to the _default_ section? Wouldn't we want to update if any section has a different set of channels?

Reply:
As of now they can only be added to the default section